keras/optimizers/loss_scale_optimizer.py

```diff
@@ -150,37 +150,87 @@ def apply(self, grads, trainable_variables=None):
                 self.build(trainable_variables)
             self.built = True
 
-        def handle_finite_grads():
-            scale = self.dynamic_scale
-            # Unscale gradients.
-            unscaled_grads = [
-                g if g is None else ops.divide(g, scale) for g in grads
-            ]
-            self.inner_optimizer.apply(
-                unscaled_grads, trainable_variables=trainable_variables
-            )
+        if backend.backend() == "tensorflow":
+            self._tf_apply(grads, trainable_variables)
+        else:
+            self._common_apply(grads, trainable_variables)
+
+    def handle_finite_grads(self, grads, trainable_variables):
```

Comment:
Why are we pulling these out to toplevels? It is slightly confusing because we also have stateless versions for jax with the same name declared inline.

Probably we should either keep them all inlined, or pull them all out, with a `stateless_` prefix for the stateless ones?

Reply:
Thanks for the feedback. The root cause here for tf is that when running with distribution strategy, the cond and the actual apply need to happen in different context (cross-replica vs per-replica). This force us to have those method available at top level so that it can be used by distribution and non-distribution code path.

For the stateless path, I can also lift the handle_x function out, and we probably should add _ to make them private methods.

Reply:
Done. Updated.

Reply:
don't see the update yet, did you forget to push?

Reply:
ops, just pushed now.